internal/appconfig/config.go

```diff
@@ -147,3 +147,10 @@ func (c *Config) DockerBuildTarget() string {
 	}
 	return c.Build.DockerBuildTarget
 }
+
+func (c Config) GetInternalPort() int {
+	if len(c.Services) > 0 {
+		return c.Services[0].InternalPort
```

Comment:
there is a trick we need to account for, the "HttpService" field may not be nil and if present it is our reference service to get and set the internal port.

Reply:
Oh, and there is also a SetInternalPort() that needs to take HttpService into account

Reply:
>there is a trick we need to account for, the "HttpService" field may not be nil and if present it is our reference service to get and set the internal port.

just so I understand what you mean @dangra 

are you saying `http_service` takes precedence over `service`? 👇 
```go
if c.HttpService != nil {
	return c.HttpService.InternalPort
}
	
if len(c.Services) > 0 {
	return c.Services[0].InternalPort
}
```

or are you saying in need to read from both services and http_service?

```go	
if len(c.Services) > 0 {
	return c.Services[0].InternalPort
}

if c.HttpService != nil {
	return c.HttpService.InternalPort
}
```

Reply:
btw, getters and setters are easy to test and some already have test cases. Would be cool to use this opportunity to add test cases for InternalPort(), there one already for SetInternalPort() but doesn't consider HttpService.

Reply:
>Oh, and there is also a SetInternalPort() that needs to take HttpService into account

@dangra can you explain this though? this seems like new behaviour and im not too confident about introducing it in this pr. (for dockerfiles)

Reply:
Oh, I was thinking on something like this:

```diff
diff --git a/internal/appconfig/setters.go b/internal/appconfig/setters.go
index efdbab52..d30baf8e 100644
--- a/internal/appconfig/setters.go
+++ b/internal/appconfig/setters.go
@@ -16,7 +16,9 @@ import (

 func (c *Config) SetInternalPort(port int) {
        c.v1SetInternalPort(port)
-       if len(c.Services) > 0 {
+       if c.HttpService != nil {
+               c.HttpService.InternalPort = port
+       } else if len(c.Services) > 0 {
                c.Services[0].InternalPort = port
        }
 }
```

I understand if you prefer not to in this PR, we will need it for V2 apps later anyways

Reply:
Ok I'll add it in a new PR. (so i can easily revert incase it breaks something)